gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/EncounterMapper.java

```diff
@@ -0,0 +1,327 @@
+            return ts.getValue();
+        } else if (ts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+            return CsNullFlavor.UNK.value();
+        }
+
+        return null;
+    }
+
+    private String getIVXBTSStringValue(IVXBTS ivxbts) {
+        if (ivxbts == null) {
+            return null;
+        } else if (ivxbts.getValue() != null) {
+            return ivxbts.getValue();
+        } else if (ivxbts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+            return CsNullFlavor.UNK.value();
+        }
+
+        return null;
+    }
+
+    private boolean validValue(String value) {
+        return value != null && !CsNullFlavor.UNK.value().equals(value);
+    }
+
+    private EncounterParticipantComponent getRecorder(RCMRMT030101UK04Author author) {
+        var recorder = new EncounterParticipantComponent();
+        var coding = new Coding();
+
+        coding
+            .setSystem(RECORDER_SYSTEM)
+            .setCode(RECORDER_CODE)
+            .setDisplay(RECORDER_DISPLAY);
+
+        return recorder
+            .addType(new CodeableConcept(coding))
+            .setIndividual(new Reference(PRACTITIONER_REFERENCE_PREFIX + author.getAgentRef().getId().getRoot()));
+    }
+
+    private boolean isNonNullParticipant2(RCMRMT030101UK04Participant2 participant2) {
+        return participant2.getNullFlavor() == null;
+    }
+
+    private EncounterParticipantComponent getPerformer(RCMRMT030101UK04Participant2 participant2) {
+        var performer = new EncounterParticipantComponent();
+        var coding = new Coding();
+
+        coding
+            .setSystem(PERFORMER_SYSTEM)
+            .setCode(PERFORMER_CODE)
+            .setDisplay(PERFORMER_DISPLAY);
+
+        return performer
+            .addType(new CodeableConcept(coding))
+            .setIndividual(new Reference(PRACTITIONER_REFERENCE_PREFIX + participant2.getAgentRef().getId().getRoot()));
+    }
+
+    private List<EncounterParticipantComponent> getEncounterParticipant(RCMRMT030101UK04Author author,
+        List<RCMRMT030101UK04Participant2> participant2List) {
+        List<EncounterParticipantComponent> participantList = new ArrayList<>();
+
+        if (author.getNullFlavor() == null) {
+            participantList.add(getRecorder(author));
+        }
+
+        // TODO: If author has a nullFlavor then create a recorder which references the Unknown Practitioner (NIAD-2026)
+
+        var participant2 = participant2List.stream().filter(this::isNonNullParticipant2).findFirst();
+        if (participant2.isPresent()) {
+            participantList.add(getPerformer(participant2.get()));
+        }
+
+        return participantList;
+    }
+
+    private void setEncounterLocation(Encounter encounter, RCMRMT030101UK04EhrComposition ehrComposition) {
+        if (ehrComposition.getLocation() != null) {
+            var location = new EncounterLocationComponent();
+            location.setLocation(new Reference(LOCATION_REFERENCE_PREFIX
```

Comment:
Instead of having prefix and suffix as a separate constants and joining them using `+`, you could have one constant `LOCATION_REFERENCE =  "Location/%s-LOC"`, and use `.formatted(ehrComposition.getId().getRoot())` method on it.